virttest/virsh.py

```diff
@@ -621,6 +624,110 @@ def kosher_args(remote_ip, uri):
         return True not in all_false
 
 
+class EventNotFoundError(Exception):
+    """
+    Error when certain event cannot be found.
+    """
+
+    def __init__(self, details=''):
+        self.details = details
+
+    def __str__(self):
+        return str(self.details)
+
+
+class EventTracker(object):
+
+    @staticmethod
+    def start_get_event(vm_name):
+        """
+        Use a virsh session with subcommand 'event' to catch events
+
+        :param vm_name: name of the vm to be catched
+        :return: the virsh session with 'event'
+        """
+        virsh_session = aexpect.ShellSession(VIRSH_EXEC)
+        event_cmd = 'event %s --all --loop' % vm_name
+        logging.info('Sending "%s" to virsh shell', event_cmd)
+        virsh_session.sendline(event_cmd)
+
+        return virsh_session
+
+    @staticmethod
+    def finish_get_event(virsh_session):
+        """
+        Stop virsh session and return the event output
+
+        Usage:
+        virsh_session = start_get_event(vm_name)
+        ####
+        virsh commands or other operations
+        ####
+        event_output = finish_get_event(virsh_session)
+
+        :param virsh_session: virsh session to catch events
+        :return: actual event output catched
+        """
+        virsh_session.send_ctrl("^C")
+        time.sleep(5)
```

Comment:
After talk with haizhao with the sleep time here, it is acceptable as it will not blocking the session get the output as it is just to wait the event coming out totally. And it is also used in other place for waiting the event.

Reply:
ok